src/core/public/chrome/ui/header/right_navigation_button.tsx

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { EuiHeaderSectionItemButton, EuiIcon } from '@elastic/eui';
+import React from 'react';
+import { InternalApplicationStart } from '../../../application';
+import { HttpStart } from '../../../http';
+
+export interface RightNavigationButtonProps {
+  application: InternalApplicationStart;
+  http: HttpStart;
+  appId: string;
+  iconType: string;
+  title: string;
+}
+
+export const RightNavigationButton = ({
+  application,
+  http,
+  appId,
+  iconType,
+  title,
+}: RightNavigationButtonProps) => {
+  const navigateToApp = () => {
```

Comment:
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/src/core/public/chrome/ui/header/nav_link.tsx#L79

I think we keep the same logic here as left navigation.

1. When clicking, it will use SPA routing to jump to the target App.
2. When clicking combined with `metaKey`(command in Mac), it will open a new browser tab to load the app.

Reply:
Updated.